### ISSUE TITLE
GAIA-X Compliant Self-Descriptions

### ISSUE THREAD

**Issue by zeier** [agera-edc/MinimumViableDataspace#171]
## Description

Self-description attributes need to be implemented as Verifiable Presentations and need to be retrievable from an Identity Hub. 

Prototype to validate self-descriptions: https://compliance.gaia-x.eu/api/

Sample self-description (as of 7.6.2022):

```json
{
    "selfDescription": {
        "@context": {
            "sh": "http://www.w3.org/ns/shacl#",
            "xsd": "http://www.w3.org/2001/XMLSchema#",
            "gx-participant": "http://w3id.org/gaia-x/participant#"        },
        "@type": "gx-participant:LegalPerson",
        "@id": "http://example.org/participant-123456",
        "gx-participant:registrationNumber": {
            "@type": "xsd:string",
            "@value": "DEANY1234NUMBER"        },
        "gx-participant:legalAddress": {
            "@type": "gx-participant:Address",
            "gx-participant:country": {
                "@type": "xsd:string",
                "@value": "FR"            }
        },
        "gx-participant:headquarterAddress": {
            "@type": "gx-participant:Address",
            "gx-participant:country": {
                "@type": "xsd:string",
                "@value": "FR"            }
        }
    },
    "proof": {
        "created": "2022-04-21T15:59:15Z",
        "proofPurpose": "assertionMethod",
        "proofValue": "zGwuJiP42jZq2sMFG8K9t2x9dJX73Ky6pek8pUki6yuDqriQgLUVDxGyVhLY2TNwe656BB4qaZZE5SeqCWeMVxWs",
        "type": "Ed25519Signature2020",
        "verificationMethod": "did:web:vc.gaia-x.eu:issuer#Ed25519VerificationKey2020"    }
}
```

**Comment by marcgs** [comment]
* Must self-descriptions be retrievable from the Identity Hub? According to the [registration service specs](https://github.com/Metaform/mvd/blob/main/registration-service/registration-service-tech-spec.md#flow-sequence) this seems to be the other way around.
* Must or can self-descriptions be publicly available?

**Comment by marcgs** [comment]
Input from Peter Koen: "The Self-Description document will eventually be implemented as Verifiable Presentations coming from the Identity Hub. However, due to the current state of the Gaia-X compliance service we need to provide all Self-Descriptions in a URL to a single .json file. Therefor, we will implement the self description as a static .json file that is located in parallel to the DID in the well-known directory. Let’s call it “self-description.json” for simplicity."

--> This story is no longer needed for MVD.

**Comment by marcgs** [comment]
Input from Peter Koen: "The Self-Description document will eventually be implemented as Verifiable Presentations coming from the Identity Hub. However, due to the current state of the Gaia-X compliance service we need to provide all Self-Descriptions in a URL to a single .json file. Therefor, we will implement the self description as a static .json file that is located in parallel to the DID in the well-known directory. Let’s call it “self-description.json” for simplicity."

--> This story is no longer needed for MVD.
